src/Lecture2.hs

```diff
@@ -150,15 +171,43 @@ You're free to define any helper functions.
        treasure besides gold (if you already haven't done this).
 -}
 
+newtype Health = MkHealth Int
```

Comment:
I see lots of warnings about unused constructor names and types. I believe, you can fix this for now by exporting all these types and their constructors (if you find warnings annoying). So, similar to the `Knight (..)` line in the `module` list of exports in the beginning of the file, you can just add:

```
    , Health (..)
    , Dragon (..)
```

And so on 📦